### PR TITLE
Introduce map-activity to the Datastore

### LINKED ISSUE #382: Improve map pool file
# Problem
Currently, the `last_active` and `next_map` are saved to the map pools file. Because of this, all edits you do to it gets reset if the map in a rotation changes or the server switches to another map pool. This makes it very awkward to do any updates to the map pools without shutting down the server first.

# How to fix

Ultimately, you want to be able to edit the map pool, then the edits will be active next restart.

A suggested fix for this would be to store `last_active` and `next_map` in it's own file.

### PR DESCRIPTION
**Pull request by applenick**
# Introduce map-activity to the Datastore

Updated: April 9th, 2020

This fix will resolve #382.

After advice given by @Electroid, I made this PR work by converting the changes to make use to the `Datastore`. This should work seamlessly with existing server setups, as I tested with a used `pgm.db` and a newly generated one and both operated as intended.

As always suggestions are welcome and just let me know if anything should be adjusted and I’ll get right on it. 👍 

Signed-off-by: applenick <applenick@users.noreply.github.com>

### CODE REVIEW

**Comment by Electroid** [comment]
Instead of storing this in a flat file, we should use `Datastore`

**Comment by applenick** [comment]
@Electroid this is why feedback is great. Completely forgot about the Datastore haha 🤦‍♂. I’ll switch this PR to make use of the Datastore instead. Thanks so much!

**Comment by applenick** [comment]
@Electroid this should be good to go now!
I made use of the `Datastore` as suggested, hopefully the way I implemented the sql statements were how you wanted. I did perform testing and everything works, however let me know if you want anything adjusted.

File: core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package tc.oc.pgm.api.map;
+
+import javax.annotation.Nullable;
+
+/* Represents data related to persisting map pool info between server restarts */
+public interface PoolActivity {
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java:6]: `MapActivity`

File: core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   * @return The name of the map pool
+   */
+  public String getPoolName();
+
+  /**
+   * For rotation based pools, will be the map that is resumed upon server restart.
+   *
+   * @return The name of a map
+   */
+  public String getNextMap();
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java:20]: `getMapName`

File: core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  /**
+   * Whether this map pool was the last active. If the map pool was the last active, it will be
+   * resumed upon server restart.
+   *
+   * @return Whether the map pool was last active
+   */
+  public boolean isLastActive();
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java:28]: `isActive`

File: core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  /**
+   * Updates the map pool activity in the {@link Datastore}
+   *
+   * @param nextMap Name of next map if rotation based.
+   * @param active If this pool is the currently active one
+   */
+  public void updatePool(@Nullable String nextMap, boolean active);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/api/map/PoolActivity.java:36]: `update`

**Pull request merged**